src/connection.js

```diff
@@ -51,11 +50,24 @@ class Connection extends EventEmitter {
       throw new TypeError('Invalid server: ' + config.server);
     }
 
+    let authProvider;
+    if (config.authProvider) {
+      if (config.authProvider.ModuleSupported) {
+        authProvider = new NativeAuthProvider(this);
+      }
+      //TODO: throw error if there is no authprovider to handle
+    } else if (config.domain) {
+      authProvider = new NTLMAuthProvider(this);
```

Comment:
I've removed the `option` passed to authProviders. For each authentication mechanism, the options needed are different and huge.

Reply:
@v-suhame I just added it back. I think it was my fault I didn't explain why I added this option in the first place. 😣

The idea behind provider specific options is to decouple the `Connection` class from the providers. Without a provider specific option set, each option that a provider could potentially support would need to be stored in the `Connection` class - and `Connection` is already doing and knowing too much.

If you've any concerns about this, please let me know! 🙇

Reply:
Got it! Sorry for making you re-work :pensive: